### PR TITLE
IGNITE-12209: Transactions system view.

### PR DESCRIPTION
**Pull request by nizhikov**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteInternalTx.java [hunk 1]
```diff
@@ -44,7 +45,7 @@
 /**
  * Transaction managed by cache ({@code 'Ex'} stands for external).
  */
-public interface IgniteInternalTx {
+public interface IgniteInternalTx extends TransactionView {
```
Comment (agura) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteInternalTx.java]: Please, prefer composition over inheritance. Actually `IgniteInternalTx` is not `TransactionView`. 
Composition is also consistent with other existing  view implentations.

Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteInternalTx.java]: Fixed.

File: modules/core/src/main/java/org/apache/ignite/spi/systemview/view/TransactionView.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+/**
+ * Transaction representation for a {@link SystemView}.
+ */
+public class TransactionView {
+    /** Transaction. */
+    private final IgniteInternalTx tx;
+
+    /**
+     * @param tx Transaction;
```
Comment (dgarus) [inline modules/core/src/main/java/org/apache/ignite/spi/systemview/view/TransactionView.java]: type in JavaDoc ';' -> '.'

File: modules/core/src/main/java/org/apache/ignite/spi/systemview/view/TransactionView.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+     */
+    public TransactionView(IgniteInternalTx tx) {
+        this.tx = tx;
+    }
+
+    /**
+     * ID of the node on which this transaction started.
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/spi/systemview/view/TransactionView.java]: Seems, we can't.

File: modules/core/src/test/java/org/apache/ignite/internal/metric/SystemViewSelfTest.java [hunk 4]
```diff
@@ -450,6 +464,121 @@ public void testClientsConnections() throws Exception {
...
+                    catch (InterruptedException e) {
+                        throw new RuntimeException(e);
+                    }
+                }, 5, "xxx");
+
+                boolean res = waitForCondition(() -> txs.size() == 5, 5_000L);
+
+                assertTrue(res);
+
+                {
```
Comment (dgarus) [inline modules/core/src/test/java/org/apache/ignite/internal/metric/SystemViewSelfTest.java]: Why do we need that as a separate block of code?

File: modules/core/src/main/java/org/apache/ignite/spi/systemview/view/TransactionView.java [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+    public String topVer() {
+        return Objects.toString(tx.topologyVersion());
+    }
+
+    /** @return Duration in millis. */
+    public long duration() {
+        return U.currentTimeMillis() - tx.startTime();
+    }
+
+    /** @return Keys count. */
```
Comment (dgarus) [inline modules/core/src/main/java/org/apache/ignite/spi/systemview/view/TransactionView.java]: It's not clear for me what does the "Keys count" mean? Could you please, give more descriptive JavaDoc.

File: modules/core/src/main/java/org/apache/ignite/spi/systemview/view/TransactionView.java [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+    private final IgniteInternalTx tx;
+
+    /**
+     * @param tx Transaction.
+     */
+    public TransactionView(IgniteInternalTx tx) {
+        this.tx = tx;
+    }
+
+    /** @return Local node ID. */
```
[the remark this answers is not in the document]
Comment (nizhikov) [reply modules/core/src/main/java/org/apache/ignite/spi/systemview/view/TransactionView.java]: Fixed

**Pull request merged**